### DISCUSSION TITLE
Need more detailed information for deployed aspire dashboard

### DISCUSSION THREAD

**Discussion by Kumima** [microsoft/aspire#4364] [Q&A] [answered]
Currently deployed aspire dashboard hides so many details about how it's implemented by azure container apps. They should be documented or there should be a way to view the detail deployed aspire dashboard.
The list shows what's I've learned(possibly wrong) for deployed dashboard and the detail I want to explore.

1. Aspire Dashboard is provisioned as a child resource of the container app. According to the document, we need to config `azd config set alpha.aspire.dashboard on`. But this config seems not alpha anymore. And the dashboard is deployed by default, is there any way to easily disable it before provision? To edit the bicep files seems a way, but not comfortable enough.
2. Currently there is only option on azure dashboard to switch on/off the dashboard for the environment. There is no way to view the Aspire Dashboard resource detail such as memory/CPU/request usage. If I'm not wrong, dashboard is also a container which brings cost.
3. When Aspire Dashboard is provisioned, four OTLP endpoints are configured, and there is no way to view how they are configured in detail.
4. The deployed Aspire Dashboard seems using microsoft OIDC as authentication way which is not the default. This should be documented.

All the above is caused by the fact that we can't view Aspire Dashboard resource detail from azure. Is it because the `aspire-dashboard dotnetComponent` is still under development?

We need to get access to view the Aspire Dashboard resource detail although it may be the internal implementation detail. At least we should be able to view the cost caused by dashboard.

**Comment by davidfowl** [comment, marked as answer]
The dashboard in ACA is an Azure feature. You don't get to control any details other than what is publicly documented an exposed via the ARM API. You don't get to pick or configure how it is configured and that is by design. As a customer of azure, it's a black box (it could be an azure specific fork for all you know 😉). 

That said, I think it is fine to document in the ACA documentation the aspire dashboard and what it supports from a public API POV.

> Aspire Dashboard is provisioned as a child resource of the container app. According to the document, we need to config azd config set alpha.aspire.dashboard on. But this config seems not alpha anymore. And the dashboard is deployed by default, is there any way to easily disable it before provision? To edit the bicep files seems a way, but not comfortable enough.

There isn't. Can you file an issue here https://github.com/Azure/azure-dev. 

> Currently there is only option on azure dashboard to switch on/off the dashboard for the environment. There is no way to view the Aspire Dashboard resource detail such as memory/CPU/request usage. If I'm not wrong, dashboard is also a container which brings cost.

You're not paying for it.

> When Aspire Dashboard is provisioned, four OTLP endpoints are configured, and there is no way to view how they are configured in detail.

That's an implementation detail.

> The deployed Aspire Dashboard seems using microsoft OIDC as authentication way which is not the default. This should be documented.

How it works should be documented for sure.

**Comment by EdoardoTona** [comment]
> You're not paying for it.

For future readers: now the Aspire dashboard is charged like any other container (vCPU-s/GiB-s)

![image](https://github.com/user-attachments/assets/fb3b8312-e97d-42a9-ac6b-da46803c77b7)

**Comment by JasonRodman** [comment]
I too am searching for more information since I am actively trying to find a way to get it to work. My ACA environment uses private endpoints which break the aspire dashboard since its not accessible thru its auto-generated url anymore. Been searching for a workaround discussed here: https://github.com/dotnet/aspire/discussions/6960
